### ISSUE TITLE
no-unresolved for react native

### ISSUE THREAD

**Issue by flipchart** [benmosher/eslint-plugin-import#279]
In react native, you can have an import statement `import Foo from "./foo"` which could refer to the file

    foo.js
    foo.android.js
    foo.ios.js

(and I imagine they'll be adding another with Windows Universal platform support coming). 

`no-unresolved` gives an error when `foo.js` doesn't exist, but `foo.android.js` and `foo.ios.js` does. I think that it should accept the platform specific files if they exist, but perhaps could warm if one or more are missing (i.e. you have `foo.android.js` but not `foo.ios.js`)

**Comment by benmosher** [comment]
I think you may be able to do this by setting the following in an `.eslintrc`:

```yaml
settings:
  import/resolver:
    node:
      extensions:
        # if unset, default is just '.js', but it must be re-added explicitly if set 
        - .js
        - .android.js
        - .ios.js
```

See the [Node resolver README](https://www.npmjs.com/package/eslint-import-resolver-node) for more details.

Also see #283; if this works, would be good to capture in a shared config. (PR welcome! 😎)

> but perhaps could warm if one or more are missing (i.e. you have foo.android.js but not foo.ios.js)

It won't do this, it will only ensure at least one of them exists. I can imagine a rule that could enforce that all of some list exist, though. That would be cool.

**Comment by flipchart** [comment]
Thanks @benmosher the `.eslintrc` settings resolved the issue. I'll look into creating the shared config and the rule

**Comment by benmosher** [comment]
Sweet, I'ma close this and we can track via #283 or some new issue.

**Comment by sturmenta** [comment]
For using react-native-web, this works good 🔥🔥🔥

```
 "import/no-unresolved": [
      2,
      {
        "ignore": [
          "react-native"
        ]
      }
    ],
```

**Comment by jchook** [comment]
As a temporary workaround until this gets resolved:

```js
module.exports = {
 settings: {
    'import/ignore': ['react-native'],
  }
}
```
